flink-core/src/main/java/org/apache/flink/util/CollectionUtil.java

```diff
@@ -99,4 +100,18 @@ public static boolean isNullOrEmpty(Map<?, ?> map) {
 		iterable.iterator().forEachRemaining(list::add);
 		return list;
 	}
+
+	/**
+	 * Collects the elements in the Iterator in a List. If the iterator argument is null,
+	 * this method returns an empty list.
+	 */
+	public static <E> List<E> iteratorToList(@Nullable Iterator<E> iterator) {
```

Comment:
According to `IteratorUtils.toList`, if iterator is null, should throw NPE. I think it is good to be consistent with commons-collections.

Reply:
keep the same implementation of `CollectionUtil#iterableToList`